### PR TITLE
Feat: Add ability to exclude dependencies in the requirements lock file

### PR DESCRIPTION
**Pull request by izeigerman**
This adds ability to opt-out some dependencies and prevent them from being included by the automatic dependency detection logic.

### CODE REVIEW

**Review by erindru — APPROVED** [review]
LGTM

File: sqlmesh/core/context_diff.py [hunk 1]
```diff
@@ -406,3 +394,33 @@ def text_diff(self, name: str) -> str:
         except SQLMeshError as e:
             logger.warning("Failed to diff model '%s': %s", name, str(e))
             return ""
+
+
+def _build_requirements(
```
Comment (tobymao) [inline sqlmesh/core/context_diff.py:399]: why was this moved out?

**Review by tobymao — APPROVED** [review]

**Review by georgesittas — APPROVED** [review]

File: sqlmesh/core/context_diff.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/context_diff.py:399]: Cause context_diff should only care about the final set of requirements. Because of lazy computation I had to pass both `provided_requirements` and `excluded_requirements` into the instance itself just so that the actual value is computed later. This presented additional complications when it came to creating the `no_diff` instance, since I had to pass the `excluded_requirements` even though requirements were sourced from the target environment.
